components/pd_client/src/client_v2.rs

```diff
@@ -0,0 +1,1339 @@
+
+    fn get_cluster_config(&mut self) -> Result<metapb::Cluster>;
+
+    fn get_region_and_leader(
+        &mut self,
+        key: &[u8],
+    ) -> PdFuture<(metapb::Region, Option<metapb::Peer>)>;
+
+    fn get_region(&mut self, key: &[u8]) -> Result<metapb::Region>;
+
+    fn get_region_info(&mut self, key: &[u8]) -> Result<RegionInfo>;
+
+    fn get_region_async(&mut self, key: &[u8]) -> PdFuture<metapb::Region>;
+
+    fn get_region_info_async(&mut self, key: &[u8]) -> PdFuture<RegionInfo>;
+
+    fn get_region_by_id(&mut self, region_id: u64) -> PdFuture<Option<metapb::Region>>;
+
+    fn get_region_leader_by_id(
+        &mut self,
+        region_id: u64,
+    ) -> PdFuture<Option<(metapb::Region, metapb::Peer)>>;
+
+    fn ask_split(&mut self, region: metapb::Region) -> PdFuture<pdpb::AskSplitResponse>;
+
+    fn ask_batch_split(
+        &mut self,
+        region: metapb::Region,
+        count: usize,
+    ) -> PdFuture<pdpb::AskBatchSplitResponse>;
+
+    fn store_heartbeat(
+        &mut self,
+        stats: pdpb::StoreStats,
+        store_report: Option<pdpb::StoreReport>,
+        dr_autosync_status: Option<StoreDrAutoSyncStatus>,
+    ) -> PdFuture<pdpb::StoreHeartbeatResponse>;
+
+    fn report_batch_split(&mut self, regions: Vec<metapb::Region>) -> PdFuture<()>;
+
+    fn scatter_region(&mut self, region: RegionInfo) -> Result<()>;
+
+    fn get_gc_safe_point(&mut self) -> PdFuture<u64>;
+
+    fn get_operator(&mut self, region_id: u64) -> Result<pdpb::GetOperatorResponse>;
+
+    fn update_service_safe_point(
+        &mut self,
+        name: String,
+        safe_point: TimeStamp,
+        ttl: Duration,
+    ) -> PdFuture<()>;
+
+    fn report_min_resolved_ts(&mut self, store_id: u64, min_resolved_ts: u64) -> PdFuture<()>;
+}
+
+impl PdClient for RpcClient {
+    type RequestChannel<R> = mpsc::UnboundedSender<R>;
+    type ResponseChannel<R: Debug> = CachedDuplexResponse<R>;
+
+    fn create_region_heartbeat_stream(
+        &mut self,
+    ) -> Result<(
+        Self::RequestChannel<RegionHeartbeatRequest>,
+        Self::ResponseChannel<RegionHeartbeatResponse>,
+    )> {
+        let (tx, rx) = mpsc::unbounded();
+        let response_holder = Arc::new(std::sync::Mutex::new(None));
+        let response_holder_clone = response_holder.clone();
+        let mut raw_client = self.raw_client.clone();
+        let mut reconnect = self.reconnect_tx.clone();
+        let mut requests = Box::pin(rx).map(|r| {
+            fail::fail_point!("region_heartbeat_send_failed", |_| {
+                Err(grpcio::Error::RemoteStopped)
+            });
+            Ok((r, WriteFlags::default()))
+        });
+        self.raw_client.stub().spawn(async move {
+            loop {
+                if let Err(e) = raw_client.wait_for_ready().await {
+                    warn!("failed to acquire client for RegionHeartbeat stream"; "err" => ?e);
+                    let _ = GLOBAL_TIMER_HANDLE
```

Comment:
I don't get it. What's the point of backoff? Isn't it waited in `wait_for_ready` already?